cli/command/stack/deploy.go

```diff
@@ -101,47 +209,133 @@ func updateNetworks(
 		existingNetworkMap[network.Name] = network
 	}
 
-	createOpts := types.NetworkCreate{
-		Labels: getStackLabels(namespace, nil),
-		Driver: defaultNetworkDriver,
-	}
-
-	for _, internalName := range networks {
-		name := fmt.Sprintf("%s_%s", namespace, internalName)
-
+	for internalName, createOpts := range networks {
+		name := namespace.scope(internalName)
 		if _, exists := existingNetworkMap[name]; exists {
 			continue
 		}
+
+		if createOpts.Driver == "" {
+			createOpts.Driver = defaultNetworkDriver
+		}
+
 		fmt.Fprintf(dockerCli.Out(), "Creating network %s\n", name)
 		if _, err := client.NetworkCreate(ctx, name, createOpts); err != nil {
 			return err
 		}
 	}
+
 	return nil
 }
 
-func convertNetworks(networks []string, namespace string, name string) []swarm.NetworkAttachmentConfig {
+func convertServiceNetworks(
+	networks map[string]*composetypes.ServiceNetworkConfig,
+	namespace namespace,
+	name string,
+) []swarm.NetworkAttachmentConfig {
+	if len(networks) == 0 {
+		return []swarm.NetworkAttachmentConfig{
+			{
+				Target:  namespace.scope("default"),
+				Aliases: []string{name},
+			},
+		}
+	}
+
 	nets := []swarm.NetworkAttachmentConfig{}
-	for _, network := range networks {
+	for networkName, network := range networks {
 		nets = append(nets, swarm.NetworkAttachmentConfig{
-			Target:  namespace + "_" + network,
-			Aliases: []string{name},
+			Target:  namespace.scope(networkName),
+			Aliases: append(network.Aliases, name),
 		})
 	}
 	return nets
 }
 
+func convertVolumes(
+	serviceVolumes []string,
+	stackVolumes map[string]composetypes.VolumeConfig,
+	namespace namespace,
+) ([]mount.Mount, error) {
+	var mounts []mount.Mount
+
+	for _, volumeString := range serviceVolumes {
+		var (
+			source, target string
+			mountType      mount.Type
+			readOnly       bool
+			volumeOptions  *mount.VolumeOptions
+		)
+
+		// TODO: split Windows path mappings properly
+		parts := strings.SplitN(volumeString, ":", 3)
+
+		if len(parts) == 3 {
+			source = parts[0]
+			target = parts[1]
+			if parts[2] == "ro" {
+				readOnly = true
```

Comment:
Point being that these options are supported in compose, so we need to support them here.